File: DsWorkNet/Dswork.Core/Db/EntityDao.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Dswork.Core.Page;

namespace Dswork.Core.Db
{
	/// <summary>
	/// 基类接口
	/// </summary>
	/// <typeparam name="T">对象模型</typeparam>
	/// <typeparam name="PK">主键类</typeparam>
	public interface EntityDao<T, PK>
	{
		/// <summary>
		/// 新增对象
		/// </summary>
		/// <param name="entity">需要新增的对象模型</param>
		/// <returns>返回执行结果</returns>
		int Save(T entity);

		/// <summary>
		/// 删除对象
		/// </summary>
		/// <param name="primaryKey">如果是单主键的，传入主键数据类型，如果为多主键的，可以用主键类或map</param>
		/// <returns>返回执行结果</returns>
		int Delete(PK primaryKey);

		/// <summary>
		/// 更新对象
		/// </summary>
		/// <param name="entity">需要新增的对象模型</param>
		/// <returns>返回执行结果</returns>
		int Update(T entity);

		/// <summary>
		/// 查询对象
		/// </summary>
		/// <param name="primaryKey">如果是单主键的，传入主键数据类型，如果为多主键的，可以用主键类或map</param>
		/// <returns>返回执行结果</returns>
		Object Get(PK primaryKey);
	
		/// <summary>
		/// 默认列表方法
		/// </summary>
		/// <param name="pageRequest">PageRequest.getFilters()查询参数和条件数据</param>
		/// <returns>IList</returns>
		IList<T> QueryList(PageRequest pageRequest);

		/// <summary>
		/// MyBatis默认分页方法
		/// </summary>
		/// <param name="pageRequest">PageRequest.getFilters()查询参数和条件数据</param>
		/// <returns>Page</returns>
		Page<T> QueryPage(PageRequest pageRequest);
	}
}

File: DsWorkJava/dswork.core/dswork/core/db/EntityDao.java
package dswork.core.db;

import java.io.Serializable;
import java.util.List;
import java.util.Map;

import dswork.core.page.Page;
import dswork.core.page.PageRequest;

/**
 * 基类接口
 * @author skey
 * @version 3.0
 * @param &lt;T&gt; 对象模型
 * @param &lt;PK&gt; 主键类
 */
public interface EntityDao<T, PK extends Serializable>
{
	/**
	 * 新增对象
	 * @param entity 需要新增的对象模型
	 * @return int
	 */
	public int save(T entity);

	/**
	 * 删除对象
	 * @param primaryKey 如果是单主键的，传入主键数据类型，如果为多主键的，可以用主键类或map
	 * @return int
	 */
	public int delete(PK primaryKey);

	/**
	 * 更新对象
	 * @param entity 需要更新的对象模型
	 * @return int
	 */
	public int update(T entity);

	/**
	 * 查询对象
	 * @param primaryKey 如果是单主键的，传入主键数据类型，如果为多主键的，可以用主键类或map
	 * @return Object
	 */
	public Object get(PK primaryKey);
	
	/**
	 * 默认列表方法
	 * @param filters Map&lt;String, Object&gt;查询参数和条件数据
	 * @return List&lt;T&gt;
	 */
	public List<T> queryList(Map<String, Object> filters);
	
	/**
	 * 默认列表方法
	 * @param pageRequest PageRequest.getFilters()查询参数和条件数据
	 * @return List&lt;T&gt;
	 */
	public List<T> queryList(PageRequest pageRequest);

	/**
	 * MyBatis默认分页方法
	 * @param pageRequest PageRequest.getFilters()查询参数和条件数据
	 * @return Page&lt;T&gt;
	 */
	public Page<T> queryPage(PageRequest pageRequest);
}
